xayn-ai/src/ranker/mod.rs

```diff
@@ -78,26 +80,22 @@ impl Ranker {
     ///
     /// Fails if no user interests are known.
     pub(crate) fn rank(&self, documents: &mut [Document]) -> Result<(), Error> {
-        rank(documents, &self.user_interests, self.config.neighbors())
+        rank(documents, &self.user_interests)
     }
 }
 
-fn rank(
-    documents: &mut [Document],
-    user_interests: &UserInterests,
-    neighbors: usize,
-) -> Result<(), Error> {
+fn rank(documents: &mut [Document], user_interests: &UserInterests) -> Result<(), Error> {
     if documents.is_empty() {
         return Ok(());
     }
 
-    let cois_for_docs = compute_cois_for_docs(documents, user_interests, neighbors)?;
+    let cois_for_docs = compute_cois_for_docs(documents, user_interests)?;
     let score_for_docs = compute_score_for_docs(cois_for_docs.as_slice());
 
     documents.sort_unstable_by(|a, b| {
         nan_safe_f32_cmp(
-            score_for_docs.get(&b.id).unwrap(),
             score_for_docs.get(&a.id).unwrap(),
+            score_for_docs.get(&b.id).unwrap(),
```

Comment:
this change reverses the order of the sorted results, i'm not sure if this is intended, that's why i wanted to double check?

Reply:
That's intended. The previous metric was a distance (smaller is better) whereas now we have a similarity (larger is better), which is why I think we need the order flipped.

Reply:
the metric changed, but the `calculate_score()` function which takes the changed metric and is used to compute the `score_for_docs` was not changed. this is then probably related to https://github.com/xaynetwork/xayn_ai/pull/388#discussion_r789038633.

Reply:
Good point. Is [this](https://github.com/xaynetwork/xayn_ai_research/blob/c93a032884e40196e41b5fdb6ac610937c2d0234/xayn_ai/backend/rerankers/coi/coi_reranker_3_factor.py#L1410) (minus the time decay) how we would want to implement this?

So essentially just

```rust
pos_similarity - neg_similarity
```

🤔